gobblin-core/src/main/java/gobblin/converter/filter/AvroSchemaFieldRemover.java

```diff
@@ -99,6 +99,9 @@ private Schema removeFields(Schema schema, Map<String, Schema> schemaMap) {
   }
 
   private Schema removeFieldsFromRecords(Schema schema, Map<String, Schema> schemaMap) {
+    Schema newRecord = Schema.createRecord(schema.getName(), schema.getDoc(), schema.getNamespace(), schema.isError());
```

Comment:
This is probably related to the recursion on line 110. Could try to understand why this is needed and if we can avoid it? For me, this is a dangerous and error-prone approach.